addons/payment_worldline/models/payment_provider.py

```diff
@@ -114,10 +117,13 @@ def _worldline_calculate_signature(self, method, endpoint, content_type, dt_rfc)
         :param str endpoint: The endpoint to be reached by the request.
         :param str content_type: The 'Content-Type' header of the request.
         :param datetime.datetime dt_rfc: The timestamp of the request, in RFC1123 format.
+        :param str idempotency_key: The idempotency key to pass in the request.
         :return: The calculated signature.
         :rtype: str
         """
         values_to_sign = [method, content_type, dt_rfc, f'/v2/{self.worldline_pspid}/{endpoint}']
+        if idempotency_key:
+            values_to_sign = [method, content_type, dt_rfc, f'x-gcs-idempotence-key:{idempotency_key}', f'/v2/{self.worldline_pspid}/{endpoint}']
```

Comment:
is the order important? for the sake of not repeating :)
```suggestion
            values_to_sign.append( f'x-gcs-idempotence-key:{idempotency_key}')
```

Reply:
I see, then a comment will be great not to forget the reason.
Here is another suggestion but feel free to ignore if you find yours clearer. No preference from me :)
```suggestion
        values_to_sign = [method, content_type, dt_rfc]
        if idempotency_key:
            values_to_sign.append(f'x-gcs-idempotence-key:{idempotency_key}'),
        values_to_sign.append(f'/v2/{self.worldline_pspid}/{endpoint}')
```